swift/Samples/SplitScreenContainer/Sources/SplitScreenContainerViewController.swift

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Workflow
+import WorkflowUI
+
+public final class SplitScreenContainerViewController<LeftScreenType: Screen, RightScreenType: Screen>: ScreenViewController<SplitScreenContainerScreen<LeftScreenType, RightScreenType>> {
+
+    private var leftContentViewController: ScreenViewController<LeftScreenType>? = nil
+    private lazy var leftContainerView: UIView = .init()
+    private var leftContainerViewWidthConstraint: NSLayoutConstraint?
+
+    private lazy var separatorView: UIView = .init()
+
+    private var rightContentViewController: ScreenViewController<RightScreenType>? = nil
+    private lazy var rightContainerView: UIView = .init()
+
+    required init(screen: SplitScreenContainerScreen<LeftScreenType, RightScreenType>, viewRegistry: ViewRegistry) {
+        super.init(screen: screen, viewRegistry: viewRegistry)
+    }
+
+    override public func screenDidChange(from previousScreen: SplitScreenContainerScreen<LeftScreenType, RightScreenType>) {
+        update(with: screen)
+    }
+
+    private func update(with screen: SplitScreenContainerScreen<LeftScreenType, RightScreenType>) {
+        if (separatorView.backgroundColor != screen.separatorColor) {
+            separatorView.backgroundColor = screen.separatorColor
+        }
+        
+        leftContentViewController?.update(screen: screen.leftScreen)
+        rightContentViewController?.update(screen: screen.rightScreen)
+        
+        if (leftContainerViewWidthConstraint?.multiplier != screen.ratio) {
+            updateWidthConstraints(ratio: screen.ratio, animated: true)
+        }
+    }
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        
+        leftContainerView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(leftContainerView)
+        
+        separatorView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(separatorView)
+        
+        rightContainerView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(rightContainerView)
+
+        NSLayoutConstraint.activate([
```

Comment:
Might want to convert this to do frame math in viewDidLayoutSubviews for performance reasons...

Reply:
@bizzemfrog updated to use frame math in https://github.com/square/workflow/pull/925/commits/0e5e5168b54464defcbf51d00f097815130b4878